src/app/zone-ingresses/views/ZoneIngressConfigView.vue

```diff
@@ -33,7 +33,7 @@
           <template v-else>
             <ResourceCodeBlock
               id="code-block-zone-ingress"
-              :resource="data"
+              :resource="data.config"
```

Comment:
**Issue**: This isn’t correct. The YAML shown here is the ZoneIngress resource, not the config.

```suggestion
              :resource="data"
```

Reply:
Not sure if I'm not following you or if you missed this:

https://github.com/kumahq/kuma-gui/pull/1867/files#diff-9fb9e8d988eecfed0988d235c43f9055bc6a50b7e5b79890f25915568a72183eR19

? lmk

Reply:
Hm, I see.

I think it’s a bit weird to call the ZoneIngress resource "config". What we’re doing here is effectively storing the original version of a resource as it comes from the API.

Wouldn’t the `$raw` concept in the MADR work better for this?

Reply:
Can we call this `.resource` perhaps? Config is a little unfortunate at least in code as we have a few clashes with this (notable in Zones, we have a "Config" tab showing the Zone CP config (and not the Zone or ZoneOverview resource); everywhere else we call this tab "YAML" for this reason).

Reply:
Is Zones/ZoneCPs the only place where we use the name `.config` in code?

Reply:
We currently have two places where we show a config:

- Zone CP config tab: the Zone CP config (will also be used for _the_ Zone in the new `zone` mode)
- Diagnostics view: the CP config (I think; also not sure why this is called Diagnostics)

In contrast, where we show resource YAMLs (which we _do_ sometimes call configs), we call them YAML in the UI to disambiguate.

Reply:
Just to confirm a few things:

> Diagnostics view: the CP config (I think; also not sure why this is called Diagnostics)

This one is from the `/config` endpoint so I don't think it has a `.config` property. Its the entire response that we use right?

> Zone CP config tab: the Zone CP config (will also be used for the Zone in the new zone mode)

This one is the same format response that we get from `/config` but in the subscriptions `.config` property is that right?

> (will also be used for the Zone in the new zone mode)

Do I remember right that this will need a new API endpoint with the word `/config` in it that will work like the main `/config` endpoint (ie. no `.config` property we just use the entire response)?

Reply:
> This one is from the `/config` endpoint so I don't think it has a `.config` property. Its the entire response that we use right?

Correct.

> This one is the same format response that we get from `/config` but in the subscriptions `.config` property is that right?

I’m not completely sure about this one. It’s at least extremely similar, but I can’t say if there are small or big difference between those two.

> Do I remember right that this will need a new API endpoint with the word `/config` in it that will work like the main `/config` endpoint (ie. no `.config` property we just use the entire response)?

Not quite: for _the_ Zone in `zone` mode, what we show in the Zone detail view’s `config` tab is simply the response from `/config` (i.e. the same as Diagnostics). It’s the exact same endpoint, not a new one.

Reply:
Ok gotcha, ok so I definitely want to keep the `.config` word here then, if anything I would change the augmented `.config` property I added recently here:

https://github.com/kumahq/kuma-gui/blob/a4c2a9e8083e7a80c8cfc1eb32a99f321f0d729a/src/app/zones/data/index.ts#L15

I want to use `.config` for the user-land YAML config (not `.yaml` as I don't want to use the format in the name of the thing).

The 'config' from the `/config` endpoint (and related responses/properties) are not a representation of this user-land YAML config.

I just noticed that `Zones` can have a very small user-land YAML config and even though we don't display that in the GUI as yet, if we ever do that should be in `.config`, and the current `.config` (that I recently added) needs to be renamed.

Reply:
So the thing we call Config in the UI would be something like `controlPlaneConfig` on the object and the thing we call YAML in the UI would be `config` on the object? (Note I didn’t propose the name `yaml` but `resource`)?

I can live with that even though it’s weirdly cross-wired.

Reply:
> So the thing we call Config in the UI would be something like controlPlaneConfig on the object and the thing we call YAML in the UI would be config on the object?

Yes something like that, possibly `controlPlaneConfig ` or `controlPlane.config` or I dunno I can't think of something that says "config but not supplied by YAML"

> I can live with that even though it’s weirdly cross-wired.

Great, lets resolve this one then, and then we can figure out naming/shaping the outlier in `Zone.config` later.